x-pack/plugins/task_manager/server/saved_objects/migrations.ts

```diff
@@ -13,16 +14,22 @@ import {
   SavedObjectsUtils,
   SavedObjectUnsanitizedDoc,
 } from '@kbn/core/server';
+import type {
+  RuleTaskState,
+  TrackedLifecycleAlertState,
+  WrappedLifecycleRuleState,
+} from '@kbn/alerting-state-types';
+
 import { REMOVED_TYPES } from '../task_type_dictionary';
-import { ConcreteTaskInstance, TaskStatus } from '../task';
+import { SerializedConcreteTaskInstance, TaskStatus } from '../task';
```

Comment:
We've been using the wrong type, all these years!  The `ConcreteTaskInstance` type has JS types - for example, `Date` objects for dates.  The `SerializedConcreteTaskInstance` has the serialized version, so ISO date strings for dates.  The data passed here is the _serialized_ version, not the _JS_ version.

We need the Serialized version for the new migration, so I changed the types of all the migrations.  The existing migration didn't have to deal with fields that were different between the two, so it was safe to change it.